test/functional/eval/timer_spec.lua

```diff
@@ -19,69 +19,65 @@ describe('timers', function()
   end)
 
   it('works one-shot', function()
-    command("call timer_start(50, 'MyHandler')")
-    eq(0,eval("g:val"))
-    run(nil, nil, nil, load_adjust(200))
+    eq(0,eval("[timer_start(10, 'MyHandler'), g:val][1]"))
+    run(nil, nil, nil, load_adjust(50))
     eq(1,eval("g:val"))
   end)
 
   it('works one-shot when repeat=0', function()
-    command("call timer_start(50, 'MyHandler', {'repeat': 0})")
-    eq(0,eval("g:val"))
-    run(nil, nil, nil, load_adjust(200))
+    eq(0,eval("[timer_start(10, 'MyHandler', {'repeat': 0}), g:val][1]"))
+    run(nil, nil, nil, load_adjust(50))
     eq(1,eval("g:val"))
   end)
 
 
   it('works with repeat two', function()
-    command("call timer_start(50, 'MyHandler', {'repeat': 2})")
-    eq(0,eval("g:val"))
-    run(nil, nil, nil, load_adjust(300))
+    eq(0,eval("[timer_start(10, 'MyHandler', {'repeat': 2}), g:val][1]"))
+    run(nil, nil, nil, load_adjust(50))
     eq(2,eval("g:val"))
   end)
 
   it('are triggered during sleep', function()
-    command("call timer_start(50, 'MyHandler', {'repeat': 2})")
+    eq(0,eval("[timer_start(10, 'MyHandler', {'repeat': 2}), g:val][1]"))
     nvim_async("command", "sleep 10")
-    eq(0,eval("g:val"))
-    run(nil, nil, nil, load_adjust(300))
+    run(nil, nil, nil, load_adjust(50))
     eq(2,eval("g:val"))
   end)
 
   it('works with zero timeout', function()
     -- timer_start does still not invoke the callback immediately
     eq(0,eval("[timer_start(0, 'MyHandler', {'repeat': 1000}), g:val][1]"))
-    run(nil, nil, nil, load_adjust(400))
-    eq(1000,eval("g:val"))
+    retry(nil, nil, function()
+      eq(1000,eval("g:val"))
+    end)
   end)
 
   it('can be started during sleep', function()
     nvim_async("command", "sleep 10")
     -- this also tests that remote requests works during sleep
-    eval("timer_start(50, 'MyHandler', {'repeat': 2})")
-    eq(0,eval("g:val"))
-    run(nil, nil, nil, load_adjust(300))
-    eq(2,eval("g:val"))
+    eq(0,eval("[timer_start(10, 'MyHandler', {'repeat': 2}), g:val][1]"))
+    run(nil, nil, nil, load_adjust(30))
+    retry(nil, nil, function() eq(2,eval("g:val")) end)
```

Comment:
is run() needed still?

Reply:
Not really, but it also makes no sense to check right away, if it is expected to take 20+.